### PR TITLE
Add Management API Guardian enrollments endpoint

### PR DESCRIPTION
**Pull request by tomgi**
### Changes

Add endpoint for Guardian enrollments API

### References

https://auth0.com/docs/api/management/v2#!/Guardian/get_factors
https://auth0.com/docs/api/management/v2#!/Guardian/get_enrollments_by_id
https://auth0.com/docs/api/management/v2#!/Guardian/delete_enrollments_by_id

### Testing

* [x] This change adds unit test coverage
* [ ] This change adds integration test coverage
* [x] This change has been tested on the latest version of Ruby

### Checklist

* [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
* [x] All existing and new tests complete without errors
* [x] Rubocop passes on all added/modified files
* [x] All active GitHub checks have passed

### CODE REVIEW

File: spec/lib/auth0/api/v2/guardian_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      )
+      expect { @instance.guardian_enrollment('ENROLLMENT_ID') }.not_to raise_error
+    end
+
+    it 'is expected to raise an exception when the enrollment ID is empty' do
+      expect { @instance.guardian_enrollment(nil) }.to raise_exception(Auth0::MissingParameter)
+    end
+  end
+
+  context '.delete_user' do
```
[the remark this answers is not in the document]
Comment (tomgi) [reply spec/lib/auth0/api/v2/guardian_spec.rb]: Thanks, fixed

**Comment by tomgi** [comment]
Thanks for feedback @lbalmaceda 

To be honest I left out the remaining Guardian endpoints for a purely pragmatic reason - I didn't need them in my project that uses this gem 😅 

But that's a fair point, since I'm adding some of the Guardian endpoints anyway, I might as well spend some time to add all of them - I will try to do that later today.

**Comment by tomgi** [comment]
Hey @lbalmaceda, I've added all the remaining Guardian endpoints

File: lib/auth0/api/v2/guardian.rb [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    module V2
+      # Methods to use the guardian endpoints
+      module Guardian
+        include Auth0::Mixins::Validation
+
+        # Retrieves a list of Guardian factors.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/get_factors
+        # @return [json] Returns the list of all Guardian factors.
+        def guardian_factors
+          path = "#{guardian_path}/factors"
```
Comment (joshcanhelp) [inline lib/auth0/api/v2/guardian.rb]: You have `@guardian_factors_path` defined already, might as well use that.

File: lib/auth0/api/v2/guardian.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          get(path)
+        end
+        alias get_guardian_factors guardian_factors
```
[the remark this answers is not in the document]
Comment (joshcanhelp) [reply lib/auth0/api/v2/guardian.rb]: @lbalmaceda - We have a number of aliases throughout these modules. Might be helpful for some.

File: lib/auth0/api/v2/guardian.rb [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        # Updates SMS enrollment and verification SMS templates.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/put_templates
+        # @param body [hash] The parameters to update.
+        #
+        # @return [json] Returns updated SMS enrollment and verification templates.
+        def guardian_update_enrollment_verification_templates(body)
+          raise Auth0::InvalidParameter, 'Must supply a valid body' if body.to_s.empty? || body.empty?
+
+          path = "#{guardian_factors_path}/sms/templates"
+          patch(path, body)
```
Comment (joshcanhelp) [inline lib/auth0/api/v2/guardian.rb]: Should be `put` here ... `patch` returns a 404

File: lib/auth0/api/v2/guardian.rb [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        # Updates provider configuration for AWS SNS.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/put_sns
+        # @param body [hash] The parameters to update.
+        #
+        # @return [json] Returns updated provider configuration for AWS SNS.
+        def guardian_update_sns_provider_config(body)
+          raise Auth0::InvalidParameter, 'Must supply a valid body' if body.to_s.empty? || body.empty?
+
+          path = "#{guardian_factors_path}/push-notification/providers/sns"
+          patch(path, body)
```
Comment (joshcanhelp) [inline lib/auth0/api/v2/guardian.rb]: Should be `put` here ... `patch` returns a 404

File: lib/auth0/api/v2/guardian.rb [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        # Updates provider configuration for Twilio.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/put_twilio
+        # @param body [hash] The parameters to update.
+        #
+        # @return [json] Returns updated provider configuration for Twilio.
+        def guardian_update_twillo_provider_config(body)
+          raise Auth0::InvalidParameter, 'Must supply a valid body' if body.to_s.empty? || body.empty?
+
+          path = "#{guardian_factors_path}/sms/providers/twilio"
+          patch(path, body)
```
Comment (joshcanhelp) [inline lib/auth0/api/v2/guardian.rb]: Should be `put` here ... `patch` returns a 404

File: lib/auth0/api/v2/guardian.rb [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/put_factors_by_name
+        # @param name [string] Name of Guardian factor to update.
+        # @param body [hash] The parameters to update.
+        # @return [json] Returns details of updated Guardian factor.
+        def guardian_update_factor(name, body)
+          raise Auth0::MissingParameter, 'Must supply a valid name' if name.to_s.empty?
+          raise Auth0::InvalidParameter, 'Must supply a valid body' if body.to_s.empty? || body.empty?
+
+          path = "#{guardian_factors_path}/#{name}"
+          patch(path, body)
```
Comment (joshcanhelp) [inline lib/auth0/api/v2/guardian.rb]: Should be `put` here ... `patch` returns a 404

File: lib/auth0/api/v2/guardian.rb (hunk 4 above)
Comment (joshcanhelp) [reply lib/auth0/api/v2/guardian.rb]: Make sure to update the tests for this as well 👍

File: lib/auth0/api/v2/guardian.rb [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        end
+        alias get_guardian_twillo_provider_config guardian_twillo_provider_config
+
+        # Updates provider configuration for Twilio.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/put_twilio
+        # @param body [hash] The parameters to update.
+        #
+        # @return [json] Returns updated provider configuration for Twilio.
+        def guardian_update_twillo_provider_config(body)
+          raise Auth0::InvalidParameter, 'Must supply a valid body' if body.to_s.empty? || body.empty?
```
Comment (joshcanhelp) [inline lib/auth0/api/v2/guardian.rb]: Are both `body.to_s.empty?` and `body.empty?` checks necessary? If so, should the order be swapped? This pertains to all of the ones below as well.

Comment (tomgi) [reply lib/auth0/api/v2/guardian.rb]: I wasn't sure about that, taken this pattern from

https://github.com/certsy/ruby-auth0/blob/c51be2a3fa755a790cc807c4e30c05f6da5ef880/lib/auth0/api/v2/users.rb#L108

I believe it's redundant though, left `body.to_s.empty?` only.

**Comment by tomgi** [comment]
Thanks @joshcanhelp, I've applied your suggestions.

**Pull request merged**